src/plugins/unified_doc_viewer/public/hooks/use_field_actions.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useMemo } from 'react';
+import createContainer from 'constate';
+import { copyToClipboard, IconType } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { DocViewRenderProps } from '@kbn/unified-doc-viewer/types';
+
+interface WithFieldParam {
+  field: string;
+}
+
+interface WithValueParam {
+  value: unknown;
+}
+
+interface TFieldActionParams extends WithFieldParam, WithValueParam {}
+
+export interface TFieldAction {
+  id: string;
+  iconType: IconType;
+  label: string;
+  onClick: () => void;
+}
+
+type UseFieldActionsDeps = Pick<
+  DocViewRenderProps,
+  'columns' | 'filter' | 'onAddColumn' | 'onRemoveColumn'
+>;
+
+/**
+ * Higher level hook that wraps the logic for the requires actions on a field.
+ */
+const useFieldActions = ({ columns, filter, onAddColumn, onRemoveColumn }: UseFieldActionsDeps) => {
+  return useMemo(
+    () => ({
+      addColumn: onAddColumn,
+      addFilterExist: ({ field }: WithFieldParam) => filter && filter('_exists_', field, '+'),
+      addFilterIn: ({ field, value }: TFieldActionParams) => filter && filter(field, value, '+'),
+      addFilterOut: ({ field, value }: TFieldActionParams) => filter && filter(field, value, '-'),
+      copyToClipboard,
+      removeColumn: onRemoveColumn,
+      toggleFieldColumn: ({ field }: WithFieldParam) => {
+        if (!columns) return;
+        const isFieldInTable = columns.includes(field);
+        if (isFieldInTable && onRemoveColumn) {
+          onRemoveColumn(field);
+        } else if (!isFieldInTable && onAddColumn) {
+          onAddColumn(field);
+        }
+      },
+    }),
+    [columns, filter, onAddColumn, onRemoveColumn]
+  );
+};
+
+export const [FieldActionsProvider, useFieldActionsContext] = createContainer(useFieldActions);
+
+/**
+ * This is a preset of the UI elements and related actions that can be used to build an action bar anywhere in a DocView
+ */
+export const useUIFieldActions = ({ field, value }: TFieldActionParams): TFieldAction[] => {
+  const actions = useFieldActionsContext();
+
+  return useMemo(
+    () => [
+      {
+        id: 'addFilterInAction',
+        iconType: 'plusInCircle',
+        label: actionFilterForText(value as string),
```

Comment:
The suggestion makes sense, and double-checking what is used for the table actions we don't even need to contact the value to the tooltip, as it would be different from the table actions. I will remove the need to contact the value on the tooltip (which is just shown next to the action) so that it also makes it easier to re-use these actions for the available doc views (upcoming refactoring on this).